src/types/Todo.ts

```diff
@@ -0,0 +1,12 @@
+import { User } from './User';
+
+export interface Todo {
+  id: number,
+  title: string,
+  completed: boolean,
+  userId: number,
+}
+
+export interface PreparedTodo extends Todo {
+  user: User | null;
```

Comment:
Once again - your solution works, but probably making `user` optional (instead of explicitly requiring `null` for "no user") would reduce the number of types and make your components slightly simpler